pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -0,0 +1,306 @@
+	client client.Client
+	scheme *runtime.Scheme
+	// k8sclientset holds the kube client that we can re-use for all kube objects other than custom resources.
+	k8sclientset *kubernetes.Clientset
+	// clusterServiceCIDR holds the cluster network service CIDR
+	clusterServiceCIDR string
+	// signer is a signer created from the user's private key
+	signer ssh.Signer
+	// recorder to generate events
+	recorder record.EventRecorder
+}
+
+// Reconcile reads that state of the cluster for a Windows Machine object and makes changes based on the state read
+// and what is in the Machine.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Info("reconciling", "namespace", request.Namespace, "name", request.Name)
+
+	if err := r.createUserDataSecret(); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "error creating user data secret")
+	}
+	// Fetch the Machine instance
+	machine := &mapi.Machine{}
+	provisionedPhase := "Provisioned"
+	if err := r.client.Get(context.TODO(), request.NamespacedName, machine); err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	if machine.Status.Phase == nil || *machine.Status.Phase != provisionedPhase {
+		// Phase can be nil and should be ignored by WMCO
+		// If the Machine is not in provisioned state, WMCO shouldn't care about it
+		return reconcile.Result{}, nil
+	}
+
+	// Get the IP address associated with the Windows machine.
+	if len(machine.Status.Addresses) == 0 {
+		return reconcile.Result{}, nil
+	}
+	ipAddress := ""
+	for _, address := range machine.Status.Addresses {
+		if address.Type == corev1.NodeInternalIP {
+			ipAddress = address.Address
+		}
+	}
+	if len(ipAddress) == 0 {
+		return reconcile.Result{}, nil
+	}
+
+	// Get the instance ID associated with the Windows machine.
+	providerID := *machine.Spec.ProviderID
+	if len(providerID) == 0 {
+		return reconcile.Result{}, nil
+	}
+	// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+	providerTokens := strings.Split(providerID, "/")
+	instanceID := providerTokens[len(providerTokens)-1]
+	if len(instanceID) == 0 {
+		return reconcile.Result{}, nil
+	}
+
+	// Make the Machine a Windows Worker node
+	if err := r.addWorkerNode(ipAddress, instanceID); err != nil {
+		log.Error(err, "Windows VM failed to be configured", "instanceID", instanceID)
```

Comment:
This error will be logged when return reconcile.Result{}, err
I do not think adding an extra log message here will add anything